api/controllers/task.ts

```diff
@@ -0,0 +1,110 @@
+import { Request, Response } from 'express';
+import prisma from './db_service';
+
+async function getAllTasks(req: Request, res: Response) {
+  const { query } = req.query;
+  console.log("query",query);
+  const tasks = await prisma.tasks.findMany();
+  if(tasks?.length > 0) {
+    res.status(200).json(tasks);
+  } else {
+    console.log("getAllTasks",tasks)
+    res.sendStatus(404);
+}
+}
+
+async function getTask( req: Request, res: Response){
+  const { task_id } = req.params;
+  console.log("getTask")
+
+  try{
+  const task= await prisma.tasks.findUnique({
+    where: {
+      id: parseInt(task_id),
+    },
+    select:{
+      id: true,
+      title:true,
+      description:true,
+      due_date:true,
+      created_at:true,
+      task_list_id:true,
+      task_status_id:true
+    }
+  });
+
+  res.status(200).json(task);
+} catch (db_error) {
+  res.status(404).json(db_error);
+}
+}
+
+async function createTask(req: Request, res: Response) {
+  console.log("createTask")
+    const { title, description, due_date, task_list_id, task_status_id } = req.body;
+
+    try {
+
+    const task = await prisma.tasks.create({
+      data: {
+        title: title,
+        description: description,
+        due_date: due_date,
+        created_at: new Date(),
+        task_list_id: task_list_id,
+        task_status_id: task_status_id,
+      },
+    });
+
+    res.status(201).json(task);
+  } catch (db_error) {
+    res.status(404).json(db_error);
+  }
+}
+
+async function updateTask(req: Request, res: Response) {
+  const { task_id } = req.params;
+  const { title, description, due_date, task_list_id, task_status_id } = req.body;
```

Comment:
Again I dont think is a good combination to use camelCase on methods and snake_case on variables.